curvefs/src/mds/topology/topology.cpp

```diff
@@ -1257,8 +1281,152 @@ TopoStatusCode TopologyImpl::ChooseNewMetaServerForCopyset(
     return TopoStatusCode::TOPO_OK;
 }
 
-TopoStatusCode TopologyImpl::ChooseAvailableMetaServers(
-    std::set<MetaServerIdType> *metaServers, PoolIdType *poolId) {
+TopoStatusCode TopologyImpl::GenCopysetAddrRandomBatch(
```

Comment:
This function is too long, and it does different things, you can consider splitting it.It's also easier to test